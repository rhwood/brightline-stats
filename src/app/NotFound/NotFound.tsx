import * as React from 'react';
import { ExclamationTriangleIcon } from '@patternfly/react-icons/dist/dynamic/icons/exclamation-triangle-icon'
import { Button } from '@patternfly/react-core/dist/dynamic/components/Button'
import {
  EmptyState,
  EmptyStateBody,
  EmptyStateFooter,
} from '@patternfly/react-core/dist/dynamic/components/EmptyState'
import { PageSection } from '@patternfly/react-core/dist/dynamic/components/Page'
import { useNavigate } from 'react-router-dom';

const BASE_PATH = process.env.BASE_PATH || '/';

const NotFound: React.FunctionComponent = () => {
  function GoHomeBtn() {
    const navigate = useNavigate();
    function handleClick() {
      navigate(BASE_PATH);
    }
    return (
      <Button onClick={handleClick}>Take me home</Button>
    );
  }

  return (
    <PageSection hasBodyWrapper={false}>
    <EmptyState headingLevel="h1" icon={ExclamationTriangleIcon} titleText="404 Page not found" variant="full">
      <EmptyStateBody>
        We didn&apos;t find a page that matches the address you navigated to.
      </EmptyStateBody><EmptyStateFooter>
      <GoHomeBtn />
    </EmptyStateFooter></EmptyState>
  </PageSection>
  )
};

export { NotFound };
