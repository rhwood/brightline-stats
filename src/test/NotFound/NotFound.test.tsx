/**
  * @jest-environment jsdom
  */
import { TextDecoder, TextEncoder } from 'util';
Object.assign(global, { TextDecoder, TextEncoder });

import * as React from 'react';
import { NotFound } from '@app/NotFound/NotFound';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom'
import { BrowserRouter, Route, Routes } from 'react-router-dom';

describe('NotFound tests', () => {
    test('should render NotFound component', () => {
        const { asFragment } = render(
            <BrowserRouter>
                <Routes>
                    <Route path="*" element= {<NotFound />} />
                </Routes>
            </BrowserRouter>
        );

        expect(asFragment()).toMatchSnapshot();
    });
});
