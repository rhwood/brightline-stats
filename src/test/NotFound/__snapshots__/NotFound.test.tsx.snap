// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`NotFound tests should render NotFound component 1`] = `
<DocumentFragment>
  <section
    class="pf-v6-c-page__main-section"
  >
    <div
      class="pf-v6-c-empty-state"
    >
      <div
        class="pf-v6-c-empty-state__content"
      >
        <div
          class="pf-v6-c-empty-state__header"
        >
          <div
            class="pf-v6-c-empty-state__icon"
          >
            <svg
              aria-hidden="true"
              class="pf-v6-svg"
              fill="currentColor"
              height="1em"
              role="img"
              viewBox="0 0 576 512"
              width="1em"
            >
              <path
                d="M569.517 440.013C587.975 472.007 564.806 512 527.94 512H48.054c-36.937 0-59.999-40.055-41.577-71.987L246.423 23.985c18.467-32.009 64.72-31.951 83.154 0l239.94 416.028zM288 354c-25.405 0-46 20.595-46 46s20.595 46 46 46 46-20.595 46-46-20.595-46-46-46zm-43.673-165.346l7.418 136c.347 6.364 5.609 11.346 11.982 11.346h48.546c6.373 0 11.635-4.982 11.982-11.346l7.418-136c.375-6.874-5.098-12.654-11.982-12.654h-63.383c-6.884 0-12.356 5.78-11.981 12.654z"
              />
            </svg>
          </div>
          <div
            class="pf-v6-c-empty-state__title"
          >
            <h1
              class="pf-v6-c-empty-state__title-text"
            >
              404 Page not found
            </h1>
          </div>
        </div>
        <div
          class="pf-v6-c-empty-state__body"
        >
          We didn't find a page that matches the address you navigated to.
        </div>
        <div
          class="pf-v6-c-empty-state__footer"
        >
          <button
            class="pf-v6-c-button pf-m-primary"
            data-ouia-component-id="OUIA-Generated-Button-primary-1"
            data-ouia-component-type="PF6/Button"
            data-ouia-safe="true"
            type="button"
          >
            <span
              class="pf-v6-c-button__text"
            >
              Take me home
            </span>
          </button>
        </div>
      </div>
    </div>
  </section>
</DocumentFragment>
`;
